docs/pages/07_service.adoc

```diff
@@ -0,0 +1,1092 @@
+
+*Syntax:*
+
+----
+actor <name> ([<entity> <mapping-field>]) [exports <service>, ...]
+    [realm "<realm>"]
+    [identity <identity>];
+----
+
+where the `exports` keyword is followed by a comma-separated list of services.
+
+The actor can export only unmapped services or mapped services that are linked to the same type (or subtype) of entity to which the actor is linked. 
+
+*Example:*
+
+----
+entity Person {
+    field String email ;
+}
+
+transfer PersonTransfer(Person person) {
+    field String email = salesPerson.email ;
+}
+
+service ManagerService {
+    function PersonTransfer create() {
+        // implement this body in Java
+    }
+}
+
+actor Manager exports ManagerService;
+----
+
+The example above allows the manager to create a new person. To grant permissions to external users, we first need to define entities, transport objects and services. The Manager actor in this application has no realm or identity, meaning this actor is an anonymous actor. Anonymous actors are publicly available, anyone who knows the URL of the Manager actor interface can invoke the create function. This way, the Manager actor is given access to the ManagerExport's create function, with which the user can create a PersonTransfer.
+
+*Permission graph*
+
+// https://en.wikipedia.org/wiki/Graph-based_access_control
+
+JUDO's security mechanism is based on a logical permission graph, where the entry point for external users to the application is the actor. Calling the functions exported by the actor, the user can access transfer objects. Once the user has access to a transfer object, they can call the services exported by the transfer object (and can access further transfer objects).
+
+The primary benefit of this mechanism is that it provides a direct mapping between the user's permissions and the transfer objects and functions they can access. This makes it easier to manage permissions and control access to the application.
+
+The example below shows how a transfer object can export services. Users who received the PersonTransfer object can call the delete function.  This function can only be called by the users who have access to the PersonTransfer object.
+
+*Syntax:*
+
+----
+transfer <name> (<entity> <alias>) [exports <service>, ...] [{
+    [member] ...
+}]
+----
+
+*Example:*
+
+----
+transfer PersonTransfer(Person person) exports PersonService {
+    field String email = salesPerson.email ;
+}
+
+service PersonService(Person person) {
+    function void delete() {
+        // implement this body in Java
+    }
+}
+----
+
+The complete authorization mechanism can be defined by carefully provisioning services to actors and transfer objects. Always be aware of that if a transfer object has already been received by an external user, then they are authorized to call the functions provided by the transfer object.
+
+=== Guard
+
+A guard is a boolean expression that must evaluate to true in order to continue the execution.
```

Comment:
I would also add this sentence: This defines a precondition that must be checked.

Reply:
ok